argparse_dataclass.py

```diff
@@ -215,10 +216,7 @@ def _add_dataclass_options(
         if field.default == field.default_factory == MISSING and not positional:
             kwargs["required"] = True
         else:
-            if field.default_factory != MISSING:
-                kwargs["default"] = field.default_factory()
-            else:
-                kwargs["default"] = field.default
+            kwargs["default"] = MISSING
```

Comment:
I'm not sure that I understand the logic here.

Reply:
The code could be updated so that just the fields that use default_factory use this logic.

Reply:
Another thing I noticed is before this change the default value would get processed by the 'type' parameter of the ArgumentParser. (I am not explaining this very well. I will try to write a simple example.)

Reply:
```Python
In [21]: %paste
def str_int():
    return "42"

@dataclass
class Parameters:
    answer: int = field(default_factory=str_int, metadata=dict(type=int))

params = parse_args(Parameters, [])
print(f"Parameters from argparse: {repr(params)}  ## value is an int")


params = Parameters()
print(f"Parameters: {repr(params)} ## value is a string")

## -- End pasted text --
Parameters from argparse: Parameters(answer=42)  ## value is an int
Parameters: Parameters(answer='42') ## value is a string

In [22]:

```

Reply:
```Python
In [7]: %paste
## After the update
## Behavior is consistent with directly using the dataclass
def str_int():
    return "42"

@dataclass
class Parameters:
    answer: int = field(default_factory=str_int, metadata=dict(type=int))

params = parse_args(Parameters, [])
print(f"Parameters from argparse: {repr(params)}  ## value is an string")


params = Parameters()
print(f"Parameters: {repr(params)} ## value is a string")

## -- End pasted text --
Parameters from argparse: Parameters(answer='42')  ## value is an string
Parameters: Parameters(answer='42') ## value is a string

In [8]:
```